torch/fx/experimental/rewriter.py

```diff
@@ -0,0 +1,64 @@
+import ast
+import astor
+import inspect
+import textwrap
+import copy
+from types import FunctionType
+
+class AST_Rewriter(ast.NodeTransformer):
+    '''
+    Take a FunctionType object representing a `forward` method, then
+    perform an AST rewrite to swap out nodes that are not symbolically
+    traceable with a callsite to the FX alternative.
+
+    To support swapping out an AST node, define a new `visit` method on
+    that node. For more details, see:
+    https://docs.python.org/3/library/ast.html#ast.NodeTransformer
+    '''
+
+    def rewrite(self, fn: FunctionType):
+
+        # Get the source string, align everything with the `def`, then
+        # remote unnecessary indentation
+        source_str = inspect.getsource(fn)
+        whitespace_prefix = source_str.split("def")[0]
+        split_source = source_str.split('\n')
+        aligned_source = [whitespace_prefix + s for s in split_source[1:]]
+        aligned_source.insert(0, split_source[0])
+        aligned_str = '\n'.join(aligned_source)
+        normalized_str = textwrap.dedent(aligned_str)
+
+        # Rewrite the original AST
+        source_ast = ast.parse(normalized_str)
+        dest_ast = ast.fix_missing_locations(self.visit(source_ast))
+        dest_str = astor.to_source(dest_ast)
+
+        # Pull out the compiled fucntion from the newly-created Module
+        code = compile(dest_ast, "", "exec")
+        globals_dict = {}
+        globals_dict = copy.copy(fn.__globals__)
+        exec(code, globals_dict)
+        fn_compiled = globals_dict["forward"]
```

Comment:
This works for `forward` on `Module`s, but I think this is going to break for a top-level traceable function that's not called `forward`. Can you add a test for that case? And here's something I think we can try:

```
        globals_dict = copy.copy(fn.__globals__)
        keys_before = set(globals_dict.keys())
        exec(code, globals_dict)
        new_keys = list(set(globals_dict.keys()) - keys_before)
        assert len(new_keys) == 1
        fn_compiled = globals_dict[new_keys[0]]
```

This will figure out the name(s) that were introduced into globals in the course of executing the code object. Since we're specifically executing a code object for a single object, I believe there should only be a single new key introduced here. Also this can be tested like:

```    def test_assert_free_fn(self):
        def foo(x):
            return torch.relu(x)

        traced = symbolic_trace(foo)
        # Add more checks here too
```